library/std/src/path.rs

```diff
@@ -2468,6 +2468,32 @@ impl Path {
         fs::metadata(self).is_ok()
     }
 
+    /// Returns `Ok(true)` if the path points at an existing entity.
+    ///
+    /// This function will traverse symbolic links to query information about the
+    /// destination file. In case of broken symbolic links this will return `Ok(false)`.
+    /// 
+    /// As opposed to the `exists()` method, this one doesn't silently ignore errors
+    /// unrelated to the path not existing. (E.g. it will return `Err(_)` in case of permission
+    /// denied on some of the paret directories.) 
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::path::Path;
+    /// assert!(!Path::new("does_not_exist.txt").try_exists().expect("Can't check existence of file does_not_exist.txt"));
+    /// assert!(Path::new("/root/secret_file.txt").try_exists().is_err());
+    /// ```
+    // TODO: stabilization should modify documentation of `exists()` to recommend this method
+    // instead.
+    #[unstable(feature = "path_try_exists", issue = "none")]
+    #[inline]
+    pub fn try_exists(&self) -> io::Result<bool> {
+        fs::metadata(self)
+            .map(|_| true)
+            .or_else(|error| if error.kind() == io::ErrorKind::NotFound { Ok(false) } else { Err(error) } )
```

Comment:
Why not use a match expression?
```suggestion
        match fs::metadata(self) {
            Ok(_) => Ok(true),
            Err(e) if e.kind() == io::ErrorKind::NotFound => Ok(false),
            Err(e) => Err(e),
        }
```

Reply:
Hmm, good point, I usually use combinators as they tend to be more succinct but match may be better here.

Is there a generally preferred style? Maybe others have a different opinion?

Reply:
I agree that in this case, a `match` is easier to read.

Reply:
Applied.